### DISCUSSION TITLE
Add after-show event to Dialog

### DISCUSSION THREAD

**Discussion by nateko** [primefaces/community#2080] [PrimeVue]
Similar to the existing 'after-hide' event it would be nice to have 'after-show' event. 

For our specific use case, we have a wrapper component for Dialog and would like to programmatically focus on the first input element after the dialog is shown instead of adding the autofocus attribute to each input in each dialog. As it is right now, we can't use the 'show' event because the Dialog's focus() method gets called after $emit('show') and without specifying autofocus attribute the dialog autofocuses on header buttons, which in our case is undesired (and there's no way to turn off autofocusing). 

I'm sure it would also benefit other scenarios when one needs to run some code _after_ the dialog opens when the 'show' event doesn't exactly cut it.

**Comment by azoom-julio-quiezi** [comment]
PrimeVue Dialog auto focus the first action button in header.
I am not sure if it is a feature or a bug, but there should definitely be an option to turn it off, at least.
